pkg/analyzer/detection/detection.go

```diff
@@ -99,6 +116,65 @@ func detectAndAnnotateCertifiedRHELComponents(name string, files analyzer.Files,
 	return rhelfeatures, languageComponents, nil
 }
 
+// getHardcodedRHCOSContentSets returns hard coded content sets for supported
+// RHCOS versions. When the RHCOS version is not supported, it returns empty. The
+// function assumes we are working in a RHCOS namespace, and uses
+// `/etc/os-release` to gather versions and metadata. But, if the expected
+// metadata is not found, is inconsistent or cannot be retrieved, it returns an
+// error.
+func getHardcodedRHCOSContentSets(files analyzer.Files) ([]string, error) {
+	// Get metadata from etc/os-release, ensure it exists.
+	osReleaseData, exists := files.Get("etc/os-release")
```

Comment:
`os-release` should be our source of truth for which RHCOS version we're running on, but thinking about long term impacts on the codebase:
We would have 2 places to determine what OS we're running on, `/ext/features/redhatrelease/` and this place here.
This might lead to a situation where we update one but not the other, and we would have a split-brain-situation.
Should/Could we use the code in `redhatrelease` to determine the RHCOS namespace here?

Reply:
>  This might lead to a situation where we update one but not the other, and we would have a split-brain-situation.

I think it would make more sense to update `redhatrelease` to use `/ec/os-release`, and enrich the namespace object with the additional metadata we need that only exist, to my knowledge, in `/etc/os-release`.

The code here doesn't want to determine the `namespace`. Instead, it is an extension of the namespace detection that goes to `/etc/os-release` to grab additional data. In fact, if we don't see the data we expect to see, in the format we understand, we ignore os-release and leave a warning behind.

Having this logic here, far from the actual namespace detection logic, might be a risk for maintainability because it is hard to connect the dots. But the future of namespace detection and other things is relatively unstable now that we are working with consolidation, so I think moving with what we have here is a good trade-off. What do you think?

Reply:
> Having this logic here, far from the actual namespace detection logic, might be a risk for maintainability because it is hard to connect the dots.

Yes, this was my primary concern, and I concur that we should move with the current state, but keep this in mind when/if we rework `redhatrelease`. Thanks!